projects/admin/src/app/record/detail-view/library-detail-view/library-detail-view.component.ts

```diff
@@ -14,22 +14,63 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { Component } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnInit } from '@angular/core';
 import { DetailRecord } from '@rero/ng-core/lib/record/detail/view/detail-record';
+import { Observable } from 'rxjs';
+import { RecordService } from '@rero/ng-core';
+import { UserService } from '../../../service/user.service';
 
 @Component({
   selector: 'admin-library-detail-view',
   templateUrl: './library-detail-view.component.html',
   styles: []
 })
-export class LibraryDetailViewComponent implements DetailRecord {
+export class LibraryDetailViewComponent implements DetailRecord, OnInit {
 
   /** Observable resolving record data */
   record$: Observable<any>;
 
   /** Resource type */
   type: string;
 
-  constructor() { }
+  /** linked locations */
+  locations = [];
+
+  constructor(
+    private _recordService: RecordService,
+    private _userService: UserService
+  ) { }
+
+  ngOnInit() {
+    // Load linked locations
+    this.record$.subscribe(
+      (data) => {
+        const libraryPid = data.metadata.pid;
+        this._recordService
+          .getRecords(
+            'locations',
+            `library.pid:${libraryPid}`,
+            1,
+            RecordService.MAX_REST_RESULTS_SIZE,
+            undefined,
+            undefined,
+            undefined,
```

Comment:
I'm surprised that doesn't cause an error, because arguments have to be an array and an object and there's no check of type in the method in ng-core.. But it's anyway not a good idea to use undefined.

Reply:
update `undefined` values by default ng-core values